color.go

```diff
@@ -148,10 +152,12 @@ func (c Color) Sprintln(a ...interface{}) string {
 // Spaces are added between operands when neither is a string.
 // Input will be colored with the parent Color.
 func (c Color) Sprint(a ...interface{}) string {
-	message := Sprint(a...)
+	pLock.Lock()
+	defer pLock.Unlock()
+	message := color.Sprint(a...)
 	messageLines := strings.Split(message, "\n")
 	for i, line := range messageLines {
-		messageLines[i] = color.RenderCode(c.String(), strings.ReplaceAll(line, color.ResetSet, Sprintf("\x1b[0m\u001B[%sm", c.String())))
+		messageLines[i] = color.RenderCode(c.String(), strings.ReplaceAll(line, color.ResetSet, color.Sprintf("\x1b[0m\u001B[%sm", c.String())))
```

Comment:
What is the reason for changing this to use `color` directly? PTerms internal Sprint has some logic, that is not in the `color` library. I think we should stick to `pterm.Sprintf`.

Reply:
This all revolves around how the `color` lib is NOT thread safe at all. A lot of the Data Races revolved around it and I needed to add the print lock to this function as it leverages the color lib directly so I needed to change that out since Sprintf also has a lock on it so we would get recursive locking and deadlock

Reply:
IMO I may have been a little lazy here and I was trying not to be too intrusive. The use of the color lib should be all wrapped up in one sub package so we can use it everywhere safely without this type of thing because its not super clear why this is necessary, could be a good follow up.

Reply:
> so I needed to change that out since Sprintf also has a lock on it so we would get recursive locking and deadlock

Ok, that makes sense! 

>  The use of the color lib should be all wrapped up in one sub package

Honestly, I often thought about completely switching the color library, or writing an own implementation. There are so many workarounds that we did in the past 2 years, because of it. Sadly, back in the days when PTerm was young and only created to be used in one of my projects, I used the color library to create the `Style`, `RGB`, `FgXxx` and `BgXxx` colors. So it's not easy to switch it.

Reply:
yeah makes sense. Probably part of a bigger effort, just another reason to write a custom implementation.